src/libs/actions/Link.ts

```diff
@@ -1,38 +1,30 @@
 import Onyx from 'react-native-onyx';
-import lodashGet from 'lodash/get';
-import _ from 'underscore';
 import ONYXKEYS from '../../ONYXKEYS';
 import asyncOpenURL from '../asyncOpenURL';
 import * as API from '../API';
 import * as Environment from '../Environment/Environment';
 import * as Url from '../Url';
 
-let isNetworkOffline = false;
+let isNetworkOffline: boolean | undefined = false;
 Onyx.connect({
     key: ONYXKEYS.NETWORK,
-    callback: (val) => (isNetworkOffline = lodashGet(val, 'isOffline', false)),
+    callback: (value) => (isNetworkOffline = value?.isOffline ?? false),
 });
 
-let currentUserEmail;
+let currentUserEmail: string | undefined;
 Onyx.connect({
     key: ONYXKEYS.SESSION,
-    callback: (val) => (currentUserEmail = lodashGet(val, 'email', '')),
+    callback: (value) => (currentUserEmail = value?.email ?? ''),
 });
 
-/**
- * @param {String} [url] the url path
- * @param {String} [shortLivedAuthToken]
- *
- * @returns {Promise<string>}
- */
-function buildOldDotURL(url, shortLivedAuthToken) {
-    const hasHashParams = url.indexOf('#') !== -1;
-    const hasURLParams = url.indexOf('?') !== -1;
+function buildOldDotURL(url?: string, shortLivedAuthToken?: string): Promise<string> {
+    const hasHashParams = url?.indexOf('#') !== -1;
+    const hasURLParams = url?.indexOf('?') !== -1;
 
     const authTokenParam = shortLivedAuthToken ? `authToken=${shortLivedAuthToken}` : '';
-    const emailParam = `email=${encodeURIComponent(currentUserEmail)}`;
-
-    const params = _.compact([authTokenParam, emailParam]).join('&');
+    const emailParam = `email=${encodeURIComponent(currentUserEmail ?? '')}`;
```

Comment:
> this will never be falsy - but I think that's correct, right? Without a `currentUserEmail` it'll still be `email=`

actaully `currentUserEmail` comes from Onyx where there could be no value for that meaning it will be undefined so in that case it would be `email=undefined` so I would keep default as empty string

Reply:
oh okay, as long as we are fine with always having that param there even if it's `undefined` which seems like the previous behavior. So I think we're good!